### ISSUE TITLE
Ignore attribute fields from source data

### ISSUE THREAD

**Issue by dougmet** [open-sdg/open-sdg#129]
We already ignore `Geocode` as a field that is neither a dissaggregation, nor a value. In the longer term we will pull through field-metadata from the data service. In the short term, to prepare for SDMX integration we need to ignore some more.

- `Observation status`
- `Unit multiplier`

**Comment by dougmet** [comment]
We think that this is the important line in the JavaScript

https://github.com/open-sdg/open-sdg/blob/de5e5dc2f571e1e6b48a35c63afc029ce5f4a874/_includes/assets/js/indicatorModel.js#L111

**Comment by dougmet** [comment]
@LucyGwilliamAdmin has setup a data branch for you to test this with [hiddencolumns-test](https://github.com/ONSdigital/sdg-data/tree/hiddencolumns-test) and this has a corresponding site branch at [hiddencolumns-test-deploy](https://github.com/ONSdigital/sdg-indicators/tree/hiddencolumns-test-deploy) you can see an indicator for which the (undesired) effect https://sdg.mango-solutions.com/hiddencolumns-test-deploy/1-1-1/

**Comment by dougmet** [comment]
![image](https://user-images.githubusercontent.com/5878305/54938692-cdb03080-4f1e-11e9-8901-273b7c6b7063.png)

**Comment by SavvasStephanides** [comment]
Thank you both @LucyGwilliamAdmin and @dougmet for setting this up! 👍

**Comment by SavvasStephanides** [comment]
Pull request here: #131 

Simple enough change, but I haven't been able to test it since `jekyll serve` is giving me problems. Awaiting review.

**Comment by dougmet** [comment]
I updated the hidden-columns-test-deploy test build 👆to use your branch. Should be able to test there

**Comment by brockfanning** [comment]
@dougmet Unfortunately I think the test server is not going to pass HTML proofing when using the latest Open SDG code, because of the change in goal page URLs.

@SavvasStephanides I tested locally and found a javasacript issue, and put up another PR with the needed addition: #133 (drop me a note if you want to trouble shoot the local Jekyll issues)

**Comment by dougmet** [comment]
Yes I messed up because I didn’t apply the 0.4.0 updates beforehand (which I have in another branch). Will rebase from that and see.

**Comment by SavvasStephanides** [comment]
Hi both. Are these issues resolved and ready to rebase?

**Comment by brockfanning** [comment]
@SavvasStephanides I believe where we are now is:
* #131 results in a javascript error because the special columns are still in the "edges" data
* #133 illustrates a javascript fix for that problem
* We may want to fix the edges problem on the sdg-build side, in which case #131 should be fine

**Comment by SavvasStephanides** [comment]
Hi @brockfanning thanks for the update. I'm not sure what you mean by "edges" data and how this results in the Javascript error.

**Comment by brockfanning** [comment]
@SavvasStephanides You can see the error happening on the [test site](https://sdg.mango-solutions.com/hiddencolumns-test-deploy/1-1-1/) for the branch. The "edgesData" property of the main object in indicatorModel.js apparently still contains the special columns, and this results in a javascript error. The quick and dirty way to avoid this is to remove the special columns from the edgesData property, which is what I've tried in #133.

Like @dougmet said in #133 we're thinking the long term goal is to handle all of this through schemas and "column metadata". So whatever we do here is a temporary fix.

As far as I can tell the remaining question is whether to go with something like #133 or to instead keep the special columns out of the edgesData by adding some code in sdg-build. (In which case #131 would be sufficient.)

**Comment by SavvasStephanides** [comment]
Thanks @brockfanning. Please excuse my ignorance but I'm still not clear what is meant by "edge data". I think a lot of this will make more sense to me if I understand what this is.

**Comment by brockfanning** [comment]
No problem at all - I never totally wrapped by head around it either, so it's helpful to try and go over it.

The [sdg-build](https://github.com/open-sdg/sdg-build) package is what takes the data (CSV) and metadata (YAML) sources and converts them into a JSON format which Open SDG is designed to expect. In the process of doing that, sdg-build [generates the edges](https://github.com/open-sdg/sdg-build/blob/master/sdg/edges.py).

Roughly speaking, these "edges" are an indication of the connections between the disaggregations. They include a "From" and a "To", indicating a "parent/child" kind of relationship. The main point of this is to account for disaggregations that are only available within other disaggregations.

For example, take [4-2-2](https://onsdigital.github.io/sdg-indicators/4-2-2/), and check out the filters on the left. You can see that:
1. County is only available when a Region is selected
2. Local authority is only available when a Region is selected
3. Region is only available when an Age is selected

Now see the very end of [this JSON data](https://onsdigital.github.io/sdg-data/comb/4-2-2.json) which sdg-build generated. This is the "edges" data:

```
"From":["Region","Region","Age"],
"To":["County","Local authority","Region"]
```

You can see that the ordering of those arrays corresponds to those parent/child (from/to) relationships above.

So getting back to the javascript error happening - it looks like when the new columns make it into these "From" and "To" arrays, it happens to cause a javascript error in some code.

@dougmet please correct me if any of that is wrong or incomplete.

**Comment by dougmet** [comment]
Hi @brockfanning yes this is exactly right. Perhaps we should put this logic on the build side (for edges) after all. It'll be super quick.

**Comment by SavvasStephanides** [comment]
Thanks @brockfanning for taking the time to explain this. It makes sense now.

**Comment by dougmet** [comment]
In sdg-build 0.3.1 we ignore these columns for edges. open-sdg/sdg-build#22

**Comment by brockfanning** [comment]
Yes, I think we can close this now, since the quick fix is now in both open-sdg and sdg-build.

As for how we would have debugged the javascript error - to be honest I probably would have started with some `console.log` statements just before line in question, and gone backwards from there. :) (realizing the actual correct approach would have been turning on developer tools in the browser and setting a breakpoint...)

**Comment by LucyGwilliamAdmin** [comment]
Will this be available in 0.5.0?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin Yep, it's merged and ready for the release.
